File: SundeeFundee/Sources/SundeeFundeeKit/DataLayer/BundledContent/BundledContentProvider.swift
import Foundation

/// Provider of bundled hardcoded content as a fallback when remote content is unavailable
public struct BundledContentProvider: Sendable, ContentClientProtocol {

    public init() {}

    public func fetchExercises() async throws -> [ContentExercise] {
        Self.exercises
    }

    public func fetchPrograms() async throws -> [ContentProgram] {
        Self.programs
    }

    public func fetchBenchmarks() async throws -> [ContentBenchmark] {
        Self.benchmarks
    }

    // MARK: - Benchmarks

    /// All bundled benchmarks mapped from BenchmarkCatalog
    public static var benchmarks: [ContentBenchmark] {
        BenchmarkCatalog.allBenchmarks.map { def in
            ContentBenchmark(
                id: def.id,
                name: def.name,
                category: def.category,
                workoutDescription: def.workoutDescription,
                scoringType: def.scoringType.rawValue,
                intensity: def.intensity?.rawValue,
                movementTags: def.movementTags,
                equipment: def.equipment,
                timeDomain: def.timeDomain,
                coachNotes: def.coachNotes,
                sortOrder: def.sortOrder,
                source: .bundled
            )
        }
    }

    // MARK: - Programs

    /// All bundled programs generated from ProgramTemplate templates
    public static var programs: [ContentProgram] {
        ProgramTemplate.allCases.enumerated().map { index, template in
            let generated = generateProgram(template: template, name: template.displayName)

            // Note: phases are not encoded to JSON since GeneratedProgramPhase is not Codable
            // The full program data can be regenerated using generateProgram() when needed
            return ContentProgram(
                id: generated.id,
                name: generated.name,
                category: generated.category,
                description: generated.description,
                durationWeeks: generated.durationWeeks,
                sessionsPerWeek: generated.sessionsPerWeek,
                difficulty: generated.difficulty,
                phases: nil,
                printablePDFURL: template.printablePDFURL,
                sortOrder: index,
                source: .bundled
            )
        }
    }

    // MARK: - Exercises

    /// All bundled exercises from the general training catalog.
    public static var exercises: [ContentExercise] {
        trainingExerciseCatalog.enumerated().map { index, entry in
            ContentExercise(
                id: entry.id,
                name: entry.id,
                category: entry.categoryLabel,
                bodyweight: entry.bodyweightOnly,
                equipment: entry.equipmentTags.map(\.rawValue),
                movementTags: [entry.movementPattern.rawValue],
                sortOrder: index,
                source: .bundled
            )
        }
    }
}

File: android/core/src/main/java/com/sundeefundee/core/data/content/BundledContentProvider.kt
package com.sundeefundee.core.data.content

import com.sundeefundee.core.data.protocol.ContentBenchmark
import com.sundeefundee.core.data.protocol.ContentClient
import com.sundeefundee.core.data.protocol.ContentExercise
import com.sundeefundee.core.data.protocol.ContentProgram

// Hardcoded content fallback — mirrors iOS BundledContentProvider
class BundledContentProvider : ContentClient {

    override suspend fun fetchExercises(): List<ContentExercise> {
        return bundledExercises
    }

    override suspend fun fetchPrograms(): List<ContentProgram> {
        return bundledPrograms
    }

    override suspend fun fetchBenchmarks(): List<ContentBenchmark> {
        return bundledBenchmarks
    }

    companion object {
        // TODO: Port all 51 weightlifting + 31 conditioning exercises from iOS ExerciseCatalog (Phase 1)
        val bundledExercises = listOf<ContentExercise>()

        // TODO: Port 7 program templates from iOS ProgramTemplateGenerator (Phase 1)
        val bundledPrograms = listOf(
            ContentProgram(
                id = "first_margarita",
                name = "The First Margarita",
                category = "strength",
                description = "8-week progressive overload program designed for cycle-aware strength training",
                durationWeeks = 8,
                sessionsPerWeek = 3,
                difficulty = "intermediate",
                isFeatured = true
            )
        )

        // TODO: Port 30 benchmarks from iOS BenchmarkCatalog (Phase 1)
        val bundledBenchmarks = listOf<ContentBenchmark>()
    }
}
